travel-event-host-app/src/app/api/register/route.ts

```diff
@@ -0,0 +1,37 @@
+import { NextResponse } from 'next/server';
+import { connectMongoDB } from '@/lib/mongodb';
+import User from '@/models/user';
+import bcrypt from 'bcrypt';
+
+export async function POST(req: Request) {
+  let { email, password, firstName, lastName, location } = await req.json();
+  console.log({ email, password });
+  email = email.toUpperCase();
+
+  let hashedPassword = '';
+
+  await connectMongoDB();
+
+  if (password) {
+    hashedPassword = await bcrypt.hash(password, 10);
+  }
```

Comment:
This is a good example of a time when we need the data coming into the API to be validated IMHO. 
I think if there is no password value in the incoming request, we should return an error response and not even create a user document in the collection with an empty string as a password. Thoughts?